demo/main.py

```diff
@@ -188,20 +190,41 @@ def body(current_demo: str):
     src = "/" + current_demo
     with me.box(
       style=me.Style(
-        width="calc(100% - 160px)",
+        width="calc(100% - 150px)",
         display="grid",
-        grid_template_columns="1fr 1fr",
+        grid_template_columns="1fr" if state.preview_fullscreen else "1fr 1fr",
       )
     ):
       demo_ui(src)
-      demo_code(inspect.getsource(get_module(current_demo)))
+      if not state.preview_fullscreen:
+        demo_code(inspect.getsource(get_module(current_demo)))
 
 
 def demo_ui(src: str):
+  state = me.state(State)
   with me.box(
     style=me.Style(flex_grow=1),
   ):
-    box_header("Preview")
+    with me.box(
+      style=me.Style(
+        display="flex",
+        justify_content="space-between",
+        align_items="center",
+        border=me.Border(bottom=BORDER_SIDE),
+      )
+    ):
+      me.text(
+        "Preview",
+        style=me.Style(
+          font_weight=500,
+          padding=me.Padding.all(8),
+          border=me.Border(),
+        ),
+      )
+      with me.content_button(type="icon", on_click=toggle_fullscreen):
+        me.icon(
```

Comment:
[nit] May want to add a tooltip with the icon. I think it's a pretty common enough icon that may be not necessary, but it's a bit more user friendly.

In terms of terminology, I think "fullscreen" would be slightly misleading from a user since this just hides the code sidebar. In YouTube, I guess this would be similar to "theater mode". But probably wouldn't work in our case. So that's maybe one reason not to include a tooltip.